### DISCUSSION TITLE
Single Track Multi Wave Color

### DISCUSSION THREAD

**Discussion by shallsaur** [katspaugh/wavesurfer.js#3187] [Q&A]
Hi,

I have a single audio track with 2 speakers and would like to indicate who is speaking by having each of their parts in different color.
From what I can find, I don't think there is an out-of-the-box feature available to achieve this.

Been trying to get something like this to work using renderFunction but I am still having issues making it work together with zoom and also calculating what time is it at a particular peak index.

It will be great if we can pass in an array of start and end time and also its color to achieve this.

**Example:**
```
multiWaveColor: [
   { waveColor: 'red', timeStamp: [[00:00:00,00:00:11], [00:00:20,00:00:25]] }, 
   { waveColor: 'blue', timeStamp: [[00:00:11,00:00:20], [00:00:25,00:00:30]] }
]
```

Thank you.

**Comment by katspaugh** [comment]
You can do this with Regions: https://wavesurfer-js.org/examples/?regions.js

**Reply by shallsaur** [reply]
Regions just adds a background to it. What I hope to achieve is different strokeStyle for each time period.

**Reply by katspaugh** [reply]
Gotcha. I also can see what you mean with the calculation of an index for a particular time position being problematic, as it renders the waveform on multiple canvases when zoomed in, and it’s currently not possible to tell which portion of the peaks array you’re rendering.
